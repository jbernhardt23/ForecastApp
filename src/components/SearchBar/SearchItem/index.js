import React from 'react';
import PropTypes from 'prop-types';
import {Text, TouchableOpacity, View, StyleSheet} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {colors} from 'react-native-elements';

const SearchItem = ({item, onItemPress, onClosePress}) => {
  return (
    <View style={styles.contianer}>
      <TouchableOpacity style={styles.item} onPress={onItemPress}>
        <Text style={styles.text}>{item}</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress={onClosePress} style={styles.icon}>
        <Icon name="close" size={16} />
      </TouchableOpacity>
    </View>
  );
};

SearchItem.propTypes = {
  item: PropTypes.string.isRequired,
  onItemPress: PropTypes.func,
  onClosePress: PropTypes.func,
};

const styles = StyleSheet.create({
  contianer: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: colors.grey5,
  },
  text: {
    fontSize: 16,
  },
  item: {
    paddingVertical: 10,
    marginStart: 15,
    flex: 1,
  },
  icon: {
    marginLeft: 'auto',
    marginEnd: 15,
  },
});

export default SearchItem;
