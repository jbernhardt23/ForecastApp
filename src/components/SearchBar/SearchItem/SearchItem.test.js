import React from 'react';
import {render, fireEvent, cleanup} from 'react-native-testing-library';
import SearchItem from '.';

afterEach(cleanup);

test('should render correctly', () => {
  const wrapper = render(<SearchItem item="Santo Domingo" />);
  expect(wrapper.toJSON()).toMatchSnapshot();
});

test('should call onItemPress', () => {
  const mockPress = jest.fn();
  const {getByTestId} = render(
    <SearchItem
      testID="search-item"
      onItemPress={mockPress}
      item="Santo Domingo"
    />,
  );
  fireEvent(getByTestId('search-item'), 'onItemPress');
  expect(mockPress).toHaveBeenCalledTimes(1);
});

test('should call onClosePress', () => {
  const mockPress = jest.fn();
  const {getByTestId} = render(
    <SearchItem
      testID="search-item"
      onClosePress={mockPress}
      item="Santo Domingo"
    />,
  );
  fireEvent(getByTestId('search-item'), 'onClosePress');
  expect(mockPress).toHaveBeenCalledTimes(1);
});
