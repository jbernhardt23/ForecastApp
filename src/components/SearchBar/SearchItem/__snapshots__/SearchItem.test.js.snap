// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e1e8ee",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "marginStart": 15,
        "opacity": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Santo Domingo
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginEnd": 15,
        "marginLeft": "auto",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Óóç
    </Text>
  </View>
</View>
`;
